### PR TITLE
Do not fail if cms.protocol is not set

### PR DESCRIPTION
**Pull request by fwyzard**

### CODE REVIEW

**Comment by fwyzard** [comment]
@cmsbuild, please test

File: git-cms-merge-topic [hunk 1]
```diff
@@ -42,13 +42,9 @@ usage() {
...
 
-# get default protocol from git config
-PROTOCOL=$(git config --get cms.protocol)
-HAS_PROTOCOL=$?
-# fallback to default (ignore "mixed")
-if [ "$HAS_PROTOCOL" -ne 0 ] || [ "$PROTOCOL" = "mixed" ]; then
-  PROTOCOL=https
-elif [ "$PROTOCOL" != "https" ] && [ "$PROTOCOL" != "ssh" ] && [ "$PROTOCOL" != "mixed" ]; then
+# get default protocol from git config, or use "mixed" by default
+PROTOCOL=$(git config --get cms.protocol || echo mixed)
```
Comment (kpedro88) [inline git-cms-merge-topic]: cms-merge-topic doesn't recognize mixed, so the default should be https

**Comment by kpedro88** [comment]
thanks @fwyzard

File: git-cms-merge-topic (hunk 1 above)
Comment (smuzaffar) [reply git-cms-merge-topic]: what will happen if it is set to mixed?
Currently PR tests jobs are failing due to this update and I have set cms.protocol to be `mixed`. Will this fail for cms-merge-topic?

Comment (kpedro88) [reply git-cms-merge-topic]: The version of the code you're using has this:
```
if [ "$HAS_PROTOCOL" -ne 0 ] || [ "$PROTOCOL" = "mixed" ]; then
  PROTOCOL=https
```

We need to keep the same functionality in this PR.

Comment (fwyzard) [reply git-cms-merge-topic]: The only check i see is
```
if [ "$PROTOCOL" = "ssh" ]; then
  REPOSITORY=git@github.com:$GITHUB_USER/cmssw.git
else
  REPOSITORY=https://github.com/$GITHUB_USER/cmssw.git
fi
```
so `mixed` should behave just as `https`.

**Comment by smuzaffar** [comment]
Thanks @fwyzard , changes look good to me. As I am starting a CXXMODULE IB, so I am merging this and including it in the IB to get this fix deployed on cvmfs.

File: git-cms-merge-topic (hunk 1 above)
Comment (kpedro88) [reply git-cms-merge-topic]: Fair point.

**Pull request merged**